src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -360,6 +356,28 @@ class IpexCommunicationService extends AgentService {
       };
     }
   }
+
+  @OnlineOnly
+  async waitAndUpdateCredential(
```

Comment:
We still need to call `getCredentialBySaid` before updating the metadata, right? Shouldn't we keep the `@OnlineOnly`?

Reply:
But this function only calls `updateAcdcMetadataRecordCompleted` and emits an event. Where is it calling `getCredentialBySaid`? I mean in _this_ function specifically

Reply:
@iFergal The `updateAcdcMetadataRecordCompleted` is taking the result of `getCredentialBySaid` as an argument

Reply:
@Sotatek-TungNguyen2a Ah, true. Can we pass instead pass the credential as an argument? Since we have it with `grantExchange.exn.e.acdc` - does it work by passing this?

Reply:
Couple of points:
- `updateAcdcMetadataRecordCompleted` is only used by that function. Can we just have one `markAcdcComplete` and join the functions?
- We just need the `credentialType` which is `cred.schema.type` - is this in the IPEX grant message?
  - Can we check if this is actually already available at the start of `acceptAcdc`? I think it should be. If so, we can save the `credentialType` at the start with an updated `saveAcdcMetadataRecord`

Reply:
> Since we have it with grantExchange.exn.e.acdc - does it work by passing this?
The value of `grantExchange.exn.e.acdc` is different with what we get from `getCredentialBySaid`, however, it contains the SAID of the credential so we can still get it but I think passing it like that won't change anything, we still need to get the credential by its SAID.

Reply:
![image](https://github.com/user-attachments/assets/f64bfb22-e98b-4e84-8974-a5609d18bf71)

> We just need the credentialType which is cred.schema.type - is this in the IPEX grant message?

I checked the `cred.schema.type` but I think the value is not what we are looking for but the `$id` key is indeed the schema SAID.

Reply:
![image](https://github.com/user-attachments/assets/3d565211-9a6e-4885-9af2-990c27560346)
The grant messgage also contains that schema SAID. But I don't get your idea at this. Do you mean we only accept the schema said that related to our `CredentialType`, else we will ignore?

Reply:
@Sotatek-TungNguyen2a Actually I meant, `cred.schema.title` - my bad. But this is what we want.

Is the schema in the IPEX grant message?

If so, we can do:
```

  async markAcdcComplete(credentialId: string, title: string) {
    const metadata = await this.credentialStorage.getCredentialMetadata(
      `metadata:${id}`
    );
    if (!metadata) {
      throw new Error(
        IpexCommunicationService.CREDENTIAL_MISSING_METADATA_ERROR_MSG
      );
    }

    metadata.status = CredentialMetadataRecordStatus.CONFIRMED;
    metadata.credentialType = title;
    await this.credentialStorage.updateCredentialMetadata(
      metadata.id,
      metadata
    );

    this.props.eventService.emit<AcdcStateChangedEvent>({
      type: AcdcEventTypes.AcdcStateChanged,
      payload: {
        status: CredentialStatus.CONFIRMED,
        credential: getCredentialShortDetails(metadata),
      },
    });
  }
```

Reply:
The field is called `credentialType` for legacy reasons, but it's what's shown on the cards in the UI in the list of credentials.

Reply:
![image](https://github.com/user-attachments/assets/618c9a75-ca0a-4473-9245-de19f1a1a903)
The grant message does not have that `title` so we can keep the current code, right?